kedro-telemetry/tests/test_plugin.py

```diff
@@ -192,6 +196,19 @@ def test_before_command_run_anonymous(self, mocker, fake_metadata):
         ]
         assert mocked_heap_call.call_args_list == expected_calls
 
+    def test_before_command_run_heap_call_error(self, mocker, fake_metadata):
+        mocker.patch(
+            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
+        )
+        mocked_heap_call = mocker.patch(
+            "kedro_telemetry.plugin._send_heap_event", side_effect=Exception
+        )
+        telemetry_hook = KedroTelemetryCLIHooks()
+        command_args = ["--version"]
+
+        telemetry_hook.before_command_run(fake_metadata, command_args)
+        mocked_heap_call.assert_called()
```

Comment:
I would explicitly check that the right exception message gets thrown. Also in the test below.